### PR TITLE
Update dependency version

### LINKED ISSUE #286: Allow user to install higher version of Panda3d, so that they can use M1/2 chip Mac to run MetaDrive (without renderer)

### LINKED ISSUE #299: Update gym
Gym 0.19.0: cloudpickle>=1.2.0,<1.7.0
Gym 0.26.2: cloudpickle>=1.2.0

We need a newer version of cloud pickle for installing metadrive as a dependency.

Even 0.20 is fine: https://github.com/openai/gym/blob/v0.20.0/requirements.txt

### LINKED ISSUE #301: Update numpy
numpy<=1.19.3

Too restrictive. Need numpy>=1.21.6

### LINKED ISSUE #303: Error gym 0.24.1
```
/home/jonch/.cache/pypoetry/virtualenvs/openpilot-P9NmkXAX-py3.8/lib/python3.8/site-packages/gym/utils/passive_env_checker.py:97: UserWarning: WARN: We recommend you to use a symmetric and normalized Box action space (range=[-1, 1]) https://stable-baselines3.readthedocs.io/en/master/guide/rl_tips.html
  logger.warn(
/home/jonch/.cache/pypoetry/virtualenvs/openpilot-P9NmkXAX-py3.8/lib/python3.8/site-packages/gym/utils/passive_env_checker.py:216: UserWarning: WARN: Future gym versions will require that `Env.reset` can be passed a `seed` instead of using `Env.seed` for resetting the environment random number generator. 
  logger.warn(
/home/jonch/.cache/pypoetry/virtualenvs/openpilot-P9NmkXAX-py3.8/lib/python3.8/site-packages/gym/utils/passive_env_checker.py:228: UserWarning: WARN: Future gym versions will require that `Env.reset` can be passed `return_info` to return information from the environment resetting.
  logger.warn(
/home/jonch/.cache/pypoetry/virtualenvs/openpilot-P9NmkXAX-py3.8/lib/python3.8/site-packages/gym/utils/passive_env_checker.py:233: UserWarning: WARN: Future gym versions will require that `Env.reset` can be passed `options` to allow the environment initialisation to be passed additional information.
  logger.warn(
Known pipe types:
  glxGraphicsPipe
(all display modules loaded.)
:device(error): Error adding inotify watch on /dev/input: No space left on device
You may using too large buffer! The width is 1208, and length is 1928.
SELF 1928 1208
Process Process-1:
Traceback (most recent call last):
  File "/home/jonch/.pyenv/versions/3.8.10/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/home/jonch/.pyenv/versions/3.8.10/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "tools/sim/bridge.py", line 372, in bridge_keep_alive
    self._run(q)
  File "tools/sim/bridge.py", line 396, in _run
    world = self.spawn_objects()
  File "tools/sim/bridge.py", line 718, in spawn_objects
    env.step([0.0, 1.0])
  File "/home/jonch/.cache/pypoetry/virtualenvs/openpilot-P9NmkXAX-py3.8/lib/python3.8/site-packages/gym/wrappers/order_enforcing.py", line 37, in step
    return self.env.step(action)
  File "/home/jonch/.cache/pypoetry/virtualenvs/openpilot-P9NmkXAX-py3.8/lib/python3.8/site-packages/gym/wrappers/env_checker.py", line 39, in step
    return passive_env_step_check(self.env, action)
  File "/home/jonch/.cache/pypoetry/virtualenvs/openpilot-P9NmkXAX-py3.8/lib/python3.8/site-packages/gym/utils/passive_env_checker.py", line 273, in passive_env_step_check
    if np.any(np.isnan(obs)):
TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

### PR DESCRIPTION
**Pull request by QuanyiLi**
## What changes do you make in this PR?

    "gym>=0.20.0, <=0.24.0",
    "numpy>=1.21.6, <=1.24.2",
    "python>=3.6, <12",
    "panda3d>=1.10.8, <=1.10.13"

close #299  
close #301 
close #286
close #303 

Unitest Pass with Panda=1.10.13, numpy=1.24.2, gym=0.24.0, python-3.11:
- [x] Linux
- [x] Windows
- [x] Mac  

@pengzhenghao Could you help test python-3.11 on Mac? Just run all unit test locally

## Checklist

* [x] I have merged the latest main branch into current branch.
* [x] I have run `bash scripts/format.sh` before merging.
* Please use "squash and merge" mode.

### CODE REVIEW

**Comment by QuanyiLi** [comment]
link #301  link#299

**Comment by QuanyiLi** [comment]
Test Pass with Panda=1.10.13, numpy=1.24.2, gym=0.24.1:
- [x] Linux
- [x] Windows
- [x] Mac

**Comment by jon-chuang** [comment]
@QuanyiLi , let me quickly check if it fixes all the dependency problems.

File: setup.py [hunk 1]
```diff
@@ -106,7 +107,7 @@ def is_win():
 
 setup(
     name="metadrive-simulator",
-    python_requires='>=3.7, <=3.9',
+    python_requires='>=3.7, <3.10',
```
Comment (jon-chuang) [inline setup.py:110]: Can it work on 3.11 too?

Comment (QuanyiLi) [reply setup.py]: Yes, it can work. Just tested it and updated the ```setup.py```

**Comment by pengzhenghao** [comment]
Testing on Mac with Python 3.11, wait me

**Comment by pengzhenghao** [comment]
@QuanyiLi All tests passed in Mac with python=3.11

Except test_waymo_memory_leak which shows the memory usage is -100MB. I guess this is because it count the memory usage in concurrent unit tests. 

I rerun the test independently after batch testing, and it passes.

**Comment by pengzhenghao** [comment]
@QuanyiLi 

I encouter such error:

```
ERROR: Package 'metadrive-simulator' requires a different Python: 3.9.12 not in '<=3.9,>=3.7'
```

We should specify **the version up to 3 numbers** instead of only use two number. A correct version should be like:

```
python<=3.9.11
```

or you can simply use:

```
python<3.10
```

This should be fixed before this PR is merged!!!!

**Comment by QuanyiLi** [comment]
> @QuanyiLi
> 
> I encouter such error:
> 
> ```
> ERROR: Package 'metadrive-simulator' requires a different Python: 3.9.12 not in '<=3.9,>=3.7'
> ```
> 
> We should specify **the version up to 3 numbers** instead of only use two number. A correct version should be like:
> 
> ```
> python<=3.9.11
> ```
> 
> or you can simply use:
> 
> ```
> python<3.10
> ```
> 
> This should be fixed before this PR is merged!!!!

@pengzhenghao Did you test on the latest commit of this branch?

**Comment by pengzhenghao** [comment]
> > @QuanyiLi
> > I encouter such error:
> > ```
> > ERROR: Package 'metadrive-simulator' requires a different Python: 3.9.12 not in '<=3.9,>=3.7'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > We should specify **the version up to 3 numbers** instead of only use two number. A correct version should be like:
> > ```
> > python<=3.9.11
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > or you can simply use:
> > ```
> > python<3.10
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This should be fixed before this PR is merged!!!!
> 
> @pengzhenghao Did you test on the latest commit of this branch?

My bad. This issue is fixed in latest commit.

**Pull request merged**